### ISSUE TITLE
Ubuntu 20.04 Steam Randomly Crashing

### ISSUE THREAD

**Issue by SergeySave** [ValveSoftware/steam-for-linux#8001]
#### Your system information

* Steam client version (build number or date): Jul 20, 2021, at 22:25:57
* Distribution (e.g. Ubuntu): Ubuntu 20.04.2 LTS
* Opted into Steam client beta?: No
* Have you checked for system updates?: Yes

#### Please describe your issue in as much detail as possible:
I recently reinstalled Ubuntu on my computer (the same version as it had been) and now steam randomly crashes all of the time. Sometimes the client will stay open for hours at a time and sometimes it only lasts a couple of minutes before it crashes. I have tried reinstalling steam multiple times (using both the `.deb` file provided on the website and the default Ubuntu apt installation). At first I wasn't seeing any errors (maybe I had been looking in the wrong place or maybe I made it worse (either way it didn't get any less stable)), but I now see a seg fault in the log file.

#### Steps for reproducing this issue:
(I haven't tried going through all of these steps again, but this is roughly what I did to get here)
1. Install Ubuntu 20.04
2. Install graphics drivers
3. Install steam

**Comment by kisak-valve** [comment]
Hello @SergeySave, if you run `steam` from a terminal, does the terminal spew give any hints when Steam crashes?

**Comment by SergeySave** [comment]
No, the terminal output is pretty much empty. The following was the entire output:
```
Running Steam on ubuntu 20.04 64-bit
STEAM_RUNTIME is enabled automatically
Steam runtime environment up-to-date!
Steam client's requirements are satisfied
WARNING: Using default/fallback debugger launch
/home/<user>/.steam/debian-installation/ubuntu12_32/steam -nominidumps -nobreakpad
[2021-08-18 17:56:15] Startup - updater built Jul 20 2021 22:25:51
[2021-08-18 17:56:15] Loading cached metrics from disk (/home/<user>/.steam/debian-installation/package/steam_client_metrics.bin)
[2021-08-18 17:56:15] Using the following download hosts for Public, Realm steamglobal
[2021-08-18 17:56:15] 1. https://cdn.akamai.steamstatic.com, /client/, Realm 'steamglobal', weight was 100, source = 'update_hosts_cached.vdf'
[2021-08-18 17:56:15] 2. https://cdn.cloudflare.steamstatic.com, /client/, Realm 'steamglobal', weight was 100, source = 'update_hosts_cached.vdf'
[2021-08-18 17:56:15] 3. http://media.steampowered.com, /client/, Realm 'steamglobal', weight was 1, source = 'baked in'
[2021-08-18 17:56:15] Verifying installation...
[2021-08-18 17:56:15] Verification complete
STEAM_RUNTIME_HEAVY: ./steam-runtime-heavy
Invalid browser dimensions: 0 x 0
Invalid browser dimensions: 0 x 4
Invalid browser dimensions: 0 x 0
```

Meanwhile `error.log` mentions a seg fault (and this time a critical glib issue).

**Comment by SergeySave** [comment]
I reinstalled steam using the `.deb` from the website and it doesn't produce an error.log. Instead it puts all that output onto the terminal, so the end of that is:
```
(steam:4946): GLib-GObject-WARNING **: /usr/src/packages/BUILD/./gobject/gsignal.c:3397: signal name `(null)-added' is invalid for instance `0xe8b1f0d8'

(steam:4946): GLib-GObject-WARNING **: g_object_notify: object class `NMDeviceWifi' has no property named `\xb8\xa4\xfa\xdf'
assert_20210819232757_1.dmp[6043]: Uploading dump (out-of-process)
/tmp/dumps/assert_20210819232757_1.dmp
assert_20210819232757_1.dmp[6043]: Finished uploading minidump (out-of-process): success = yes
assert_20210819232757_1.dmp[6043]: response: CrashID=bp-217f6b74-6106-4aad-9d3f-e707f2210819
assert_20210819232757_1.dmp[6043]: file ''/tmp/dumps/assert_20210819232757_1.dmp'', upload yes: ''CrashID=bp-217f6b74-6106-4aad-9d3f-e707f2210819''
/home/<user>/.local/share/Steam/steam.sh: line 772:  4946 Segmentation fault      (core dumped) $STEAM_DEBUGGER $DEBUGGER_ARGS "$STEAMROOT/$STEAMEXEPATH" "$@"
```

**Comment by jscul** [comment]
Hey guys, I've been getting this issue for a while now. I've tried fresh OS installs, fresh Steam installs, rebooting, basically everything. I haven't tried going into the BIOS and messing with settings there.

**System info:**
OS: Ubuntu 21.04
Type: 64-bit
Gnome: 3.38.5
I'm using Ubuntu dark mode. Reason for stating this is I see something about accessing system theme in error logs.

Memory: 15.5 GiB
Processor: Intel® Core™ i7-9700K CPU @ 3.60GHz × 8
GPU: NVIDIA Corporation TU104 [GeForce RTX 2070 SUPER]
SSD: 1.0 TB
GPU driver: "Continue using a manually installed driver"

Here are my logs.

```
jscul@jscul:~$ steam
Running Steam on ubuntu 21.04 64-bit
STEAM_RUNTIME is enabled automatically
Steam runtime environment up-to-date!
Steam client's requirements are satisfied
WARNING: Using default/fallback debugger launch
/home/jscul/.local/share/Steam/ubuntu12_32/steam
[2021-08-20 19:50:02] Startup - updater built Jul 20 2021 22:25:51
Installing breakpad exception handler for appid(steam)/version(1626824053)
libGL error: No matching fbConfigs or visuals found
libGL error: failed to load driver: swrast
SteamUpdateUI: An X Error occurred
X Error of failed request:  BadValue (integer parameter out of range for operation)
Major opcode of failed request:  151 (GLX)
Minor opcode of failed request:  3 (X_GLXCreateContext)
Value in failed request:  0x0
Serial number of failed request:  44
xerror_handler: X failed, continuing
Looks like steam didn't shutdown cleanly, scheduling immediate update check
[2021-08-20 19:50:02] Loading cached metrics from disk (/home/jscul/.local/share/Steam/package/steam_client_metrics.bin)
[2021-08-20 19:50:02] Failed to load cached hosts file (File 'update_hosts_cached.vdf' not found), using defaults
[2021-08-20 19:50:02] Using the following download hosts for Public, Realm steamglobal
[2021-08-20 19:50:02] 1. http://media.steampowered.com, /client/, Realm 'steamglobal', weight was 1, source = 'baked in'
Installing breakpad exception handler for appid(steam)/version(1626824053)
[2021-08-20 19:50:02] Checking for update on startup
[2021-08-20 19:50:02] Checking for available updates...
[2021-08-20 19:50:02] Downloading manifest: http://media.steampowered.com/client/steam_client_ubuntu12
Installing breakpad exception handler for appid(steam)/version(1626824053)
[2021-08-20 19:50:02] Download skipped: /client/steam_client_ubuntu12 version 1626824053, installed version 1626824053, existing pending version 0
[2021-08-20 19:50:02] Nothing to do
[2021-08-20 19:50:02] Verifying installation...
[2021-08-20 19:50:02] Performing checksum verification of executable files
[2021-08-20 19:50:02] Verification complete
Loaded SDL version 2.0.15-6655637
Gtk-Message: Failed to load module "gail"
Gtk-Message: Failed to load module "atk-bridge"

(steam:244317): Gtk-WARNING **: Unable to locate theme engine in module_path: "adwaita",
/usr/share/themes/Yaru-dark/gtk-2.0/main.rc:775: error: unexpected identifier `direction', expected character `}'

(steam:244317): Gtk-WARNING **: Unable to locate theme engine in module_path: "adwaita",
/usr/share/themes/Yaru-dark/gtk-2.0/hacks.rc:28: error: invalid string constant "normal_entry", expected valid string constant
libGL error: No matching fbConfigs or visuals found
libGL error: failed to load driver: swrast
Steam: An X Error occurred
X Error of failed request:  BadMatch (invalid parameter attributes)
Major opcode of failed request:  151
Serial number of failed request:  49
xerror_handler: X failed, continuing
assert_20210820195002_1.dmp[244364]: Uploading dump (out-of-process)
/tmp/dumps/assert_20210820195002_1.dmp
/home/jscul/.local/share/Steam/steam.sh: line 772: 244317 Segmentation fault      (core dumped) $STEAM_DEBUGGER $DEBUGGER_ARGS "$STEAMROOT/$STEAMEXEPATH" "$@"
jscul@jscul:~$ assert_20210820195002_1.dmp[244364]: Finished uploading minidump (out-of-process): success = yes
assert_20210820195002_1.dmp[244364]: response: CrashID=bp-3a841d74-6645-459f-a68b-2568e2210820
assert_20210820195002_1.dmp[244364]: file ''/tmp/dumps/assert_20210820195002_1.dmp'', upload yes: ''CrashID=bp-3a841d74-6645-459f-a68b-2568e2210820''
```

**Comment by kisak-valve** [comment]
Hello @jscul, `swrast` is mesa's slower software renderer, and the fact that Steam is trying to use it hints that you're using nVidia's proprietary driver, but don't have the driver's 32 bit userspace support installed. The nVidia proprietary driver requires that the running kernel module, 64 bit userspace, and 32 bit userspace versions are installed and have exactly the same version to work properly.

Please figure out what nVidia driver series you're using, then run something like `sudo apt install libnvidia-gl-470:i386` (replacing 470 with the driver series you're using) to install the missing userspace support.

As a side note, your issue is unrelated to this issue report.

**Comment by jscul** [comment]
Thanks Kisak, sorry for off-topic issue, that install fixed it!

**Comment by TTimo** [comment]
This is crashing in the glib thread according to the minidumps server, there isn't much information to glean from the report otherwise.

**Comment by SergeySave** [comment]
Ok, so what can I do to fix it or what can I do to get a more useful report?

**Comment by SergeySave** [comment]
I think I've managed to solve the issue (it didn't crash for a day and a half on the beta version and now I've switched back to the normal version). For future reference if anyone has this problem I did this by running `sudo apt install libnm-dev:i386` (though I did install and update other packages before running that command, it was the last one I ran and it had the only noticeable impact). I'll close this issue in a day or so if steam doesn't crash.
